crates/compiler/gen_wasm/src/backend.rs

```diff
@@ -1871,11 +1885,92 @@ impl<'a, 'r> WasmBackend<'a, 'r> {
         );
     }
 
+    fn expr_union_field_ptr_at_index(
+        &mut self,
+        structure: Symbol,
+        tag_id: TagIdIntType,
+        union_layout: &UnionLayout<'a>,
+        index: u64,
+        symbol: Symbol,
+        storage: &StoredValue,
+    ) {
```

Comment:
I think we should try and share code with `expr_union_at_index`. may require some refactoring.